.github/workflows/build.yml

```
@@ -101,3 +140,72 @@ jobs:
       - name: Setup tmate debug session
         uses: mxschmitt/action-tmate@v3
         if: env.RUN_TMATE
+  test:
+    name: test source - go${{ matrix.go-version }}
+    needs:
+      - diagnostics
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        go-version:
+          - "1.19"
+          - "1.20"
+    steps:
+      - id: harden-runner
+        name: Harden the runner
+        uses: step-security/harden-runner@v2
+        with:
+          egress-policy: audit
+      - id: setup-env
+        uses: cisagov/setup-env-github-action@develop
+      - uses: actions/checkout@v4
+      - id: setup-go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Lookup Go cache directory
+        id: go-cache
+        run: |
+          echo "dir=$(go env GOCACHE)" >> $GITHUB_OUTPUT
+      - name: Lookup Go module cache directory
+        id: go-modcache
+        run: |
+          echo "dir=$(go env GOMODCACHE)" >> $GITHUB_OUTPUT
+      - uses: actions/cache@v3
+        env:
+          BASE_CACHE_KEY: "${{ github.job }}-${{ runner.os }}-\
+            go${{ steps.setup-go.outputs.go-version }}-"
+        with:
+          path: |
+            ${{ steps.go-cache.outputs.dir }}
+            ${{ steps.go-modcache.outputs.dir }}
+          key: "${{ env.BASE_CACHE_KEY }}\
+            ${{ hashFiles('**/go.sum') }}"
+          restore-keys: |
+            ${{ env.BASE_CACHE_KEY }}
+      - name: Install go-junit-report
+        env:
+          PACKAGE_URL: github.com/jstemmer/go-junit-report/v2
+          PACKAGE_VERSION: ${{ steps.setup-env.outputs.go-junit-report-version }}
+        run: go install ${PACKAGE_URL}@${PACKAGE_VERSION}
+      - name: Install GoMock
+        env:
+          PACKAGE_URL: github.com/golang/mock/mockgen
+          PACKAGE_VERSION: ${{ steps.setup-env.outputs.gomock-version }}
+        run: go install ${PACKAGE_URL}@${PACKAGE_VERSION}
+      - name: Generate mocks and run tests
+        run: |
+          export GOPATH="$HOME/go/"
+          export PATH=$PATH:$GOPATH/bin
+          go generate ./
+          go test -v ./test/* | go-junit-report -set-exit-code > report.xml
+      - name: Test summary
+        uses: test-summary/action@v1
```

Is there a reason not to use version 2 here?
```suggestion
        uses: test-summary/action@v2
```